test/validate.test.ts

```diff
@@ -1,7 +1,29 @@
-import { validateFolder, getValidators, validatePackageSize } from '../lib/validate';
+import * as path from 'path';
+import { validate, validateFolder, getValidators, validatePackageSize, readFile, getSize } from '../lib/validate';
 import * as chalk from 'chalk';
+import { listFilesRelativeToFolder } from '../lib/util/files';
+import { GetFileContentOptionsType } from '../lib/models';
 
 describe('validateFolder', () => {
+    function createValidator(fileCustomiser: (filePath: string, content: string) => string) {
+        const errorSpy = jest.fn();
+
+        async function validateFolderWithCustomiser(folderPath: string): Promise<boolean> {
```

Comment:
Experimenting a bit with customising the components definition with alternative scenarios. Not sure yet if this is the way to go, but maybe it is a start.